src/tract.c

```diff
@@ -1,737 +1,737 @@
+    if (tfs_skip(iset->tfscan, file) < 0)
+      return E_FREAD;           /* skip more comments */
+    d = tfs_getfld(iset->tfscan, file, NULL, 0);
+    if (d <= TFS_EOF)           /* read the next item */
+      return (d < 0) ? E_FREAD : 0;
+    if (buf[0] == '\0')         /* check for end of file */
+      return E_ITEMEXP;         /* and for a missing item */
+    item = nim_add(iset->nimap, buf, sizeof(ITEM));
+    if (item == EXISTS) return E_DUPITEM;  /* add the new item */
+    if (item == NULL)   return E_NOMEM;    /* to the name/id map */
+    item->frq = 0;              /* clear the frequency counters */
+    item->xfq = 0;              /* (occurrence and sum of t.a. sizes) */
+    if (d < TFS_FLD)    return E_APPEXP;
+    d = tfs_getfld(iset->tfscan, file, NULL, 0);
+    if (d <  0)         return E_FREAD;
+    if (d >= TFS_FLD)   return E_FLDCNT;
+    item->app = _appcode(buf);  /* get the appearance indicator */
+    if (item->app <  0)  return E_UNKAPP;
+  }
+  return 0;                     /* return 'ok' */
+}  /* is_readapp() */
+
+/*--------------------------------------------------------------------*/
+
+int is_read (ITEMSET *iset, FILE *file)
+{                               /* --- read a transaction */
+  int  i, d;                    /* loop variable, delimiter type */
+  char *buf;                    /* read buffer */
+  ITEM *item;                   /* pointer to item */
+
+  assert(iset && file);         /* check the function arguments */
+  iset->cnt = 0;                /* initialize the item counter */
+  if (tfs_skip(iset->tfscan, file) < 0)
+    return E_FREAD;             /* skip leading comments */
+  d   = _get_item(iset, file);  /* read the first item and */
+  buf = tfs_buf(iset->tfscan);  /* get the read buffer */
+  if ((d      == TFS_EOF)       /* if at the end of the file */
+  &&  (buf[0] == '\0'))         /* and no item has been read, */
+    return 1;                   /* return 'end of file' */
+  while ((d      == TFS_FLD)    /* read the other items */
+  &&     (buf[0] != '\0'))      /* of the transaction */
+    d = _get_item(iset, file);  /* up to the end of the record */
+  if (d < TFS_EOF) return d;    /* check for a read error */
+  if ((buf[0] == '\0') && (d == TFS_FLD) && (iset->cnt > 0))
+    return E_ITEMEXP;           /* check for an empty field */
+  ta_sort(iset->items, iset->cnt); /* prepare the transaction */
+  iset->cnt = ta_unique(iset->items, iset->cnt);
+  for (i = iset->cnt; --i >= 0; ) {
+    item = nim_byid(iset->nimap, iset->items[i]);
+    item->frq += 1;             /* count the item and */
+    item->xfq += iset->cnt;     /* sum the transaction sizes */
+  }                             /* as an importance indicator */
+  return 0;                     /* return 'ok' */
+}  /* is_read() */
+
+/*--------------------------------------------------------------------*/
+
+int is_recode (ITEMSET *iset, int minfrq, int dir, int *map, int mode, int fullS)

```

Comment:
/* Makh2018: Adding mode and fullS as parameters in order to extract full support items in case of generator itemsets */